libs/execution_context/ebpf_maps.c

```diff
@@ -1179,6 +1186,200 @@ _update_stack_map_entry(
     return result;
 }
 
+static _Requires_lock_held_(ring_buffer_map->core_map.lock) void _ebpf_ring_buffer_map_signal_async_query_complete(
+    _In_ ebpf_core_ring_buffer_map_t* ring_buffer_map)
+{
+    EBPF_LOG_ENTRY();
+    // Skip if no async_contexts have ever been queued.
+    if (!ring_buffer_map->async_contexts_trip_wire) {
+        return;
+    }
+
+    ebpf_core_map_t* map = &ring_buffer_map->core_map;
+    while (!ebpf_list_is_empty(&ring_buffer_map->async_contexts)) {
+        ebpf_core_ring_buffer_map_async_query_context_t* context = EBPF_FROM_FIELD(
+            ebpf_core_ring_buffer_map_async_query_context_t, entry, ring_buffer_map->async_contexts.Flink);
+        ebpf_ring_buffer_map_async_query_result_t* async_query_result = context->async_query_result;
+        ebpf_ring_buffer_query(
+            (ebpf_ring_buffer_t*)map->data, &async_query_result->consumer, &async_query_result->producer);
+        ebpf_list_remove_entry(&context->entry);
+        ebpf_async_complete(context->async_context, sizeof(*async_query_result), EBPF_SUCCESS);
+        ebpf_free(context);
+        context = NULL;
+    }
+}
+
+static void
+_delete_ring_buffer_map(_In_ ebpf_core_map_t* map)
+{
+    EBPF_LOG_ENTRY();
+    // Free the ring buffer.
+    ebpf_ring_buffer_destroy((ebpf_ring_buffer_t*)map->data);
+
+    ebpf_core_ring_buffer_map_t* ring_buffer_map = EBPF_FROM_FIELD(ebpf_core_ring_buffer_map_t, core_map, map);
+    // Snap the async context list.
+    ebpf_list_entry_t temp_list;
+    ebpf_list_initialize(&temp_list);
+    ebpf_lock_state_t state = ebpf_lock_lock(&map->lock);
+    ebpf_list_entry_t* first_entry = ring_buffer_map->async_contexts.Flink;
+    if (!ebpf_list_is_empty(&ring_buffer_map->async_contexts)) {
+        ebpf_list_remove_entry(&ring_buffer_map->async_contexts);
+        ebpf_list_append_tail_list(&temp_list, first_entry);
+    }
+    ebpf_lock_unlock(&map->lock, state);
+    // Cancel all pending async query operations.
+    for (ebpf_list_entry_t* temp_entry = temp_list.Flink; temp_entry != &temp_list; temp_entry = temp_entry->Flink) {
+        ebpf_core_ring_buffer_map_async_query_context_t* context =
+            EBPF_FROM_FIELD(ebpf_core_ring_buffer_map_async_query_context_t, entry, temp_entry);
+        ebpf_async_complete(context->async_context, 0, EBPF_CANCELED);
+    }
+    ebpf_epoch_free(ring_buffer_map);
+}
+
+static ebpf_core_map_t*
+_create_ring_buffer_map(_In_ const ebpf_map_definition_in_memory_t* map_definition)
+{
+    ebpf_result_t result;
+    ebpf_core_ring_buffer_map_t* ring_buffer_map = NULL;
+    ebpf_ring_buffer_t* ring_buffer = NULL;
+    ebpf_core_map_t* core_map = NULL;
+
+    EBPF_LOG_ENTRY();
+
+    // allocate
```

Comment:
```suggestion
    // Allocate.
```
Or just delete the comment since it contains no words that aren't in the code line below.